### PR TITLE
Document the weak_vendors value

### PR DESCRIPTION
**Pull request by greg0ire**
See https://github.com/symfony/symfony/pull/21539

### CODE REVIEW

**Comment by greg0ire** [comment]
Code PR: https://github.com/symfony/symfony/pull/21539

File: components/phpunit_bridge.rst [hunk 1]
```diff
@@ -119,7 +119,25 @@ an arbitrary value (ex: ``320``) will make the tests fails only if a higher
 number of deprecation notices is reached (``0`` is the default value). You can
 also set the value ``"weak"`` which will make the bridge ignore any deprecation
 notices. This is useful to projects that must use deprecated interfaces for
-backward compatibility reasons.
+backward compatibility reasons, or for libraries with lots of vendors that
```
Comment (nicolas-grekas) [inline components/phpunit_bridge.rst]: Lots of is undefined to the reader, it doesn't help one understand if that applies to its use cases or not

File: components/phpunit_bridge.rst [hunk 2]
```diff
@@ -119,7 +119,25 @@ an arbitrary value (ex: ``320``) will make the tests fails only if a higher
...
+can create a sudden backlog of deprecations to fix that would otherwise prevent
```
Comment (nicolas-grekas) [inline components/phpunit_bridge.rst]: Same for sudden imho

File: components/phpunit_bridge.rst [hunk 3]
```diff
@@ -119,7 +119,25 @@ an arbitrary value (ex: ``320``) will make the tests fails only if a higher
...
+unrelated pull requests from being merged… which leads us to ``"weak_vendors"``.
```
Comment (nicolas-grekas) [inline components/phpunit_bridge.rst]: Ellipses suggest the reader understood, which might be not true yet

**Review by nicolas-grekas — COMMENTED** [review]
Should be more descriptive imho

**Comment by nicolas-grekas** [comment]
(but not longer ;) )

File: components/phpunit_bridge.rst (hunk 1 above)
Comment (greg0ire) [inline components/phpunit_bridge.rst]: I guess one dependency with many deprecation is enough indeed.

File: components/phpunit_bridge.rst (hunk 2 above)
Comment (greg0ire) [inline components/phpunit_bridge.rst]: I want to convey the fact that this can happen without any code change in the lib itself, because dependencies are not locked.

File: components/phpunit_bridge.rst (hunk 3 above)
Comment (greg0ire) [inline components/phpunit_bridge.rst]: Replaced the ellipsis with `described below` so that the reader doesn't feel like they missed something.

**Comment by greg0ire** [comment]
I reworked it a bit. Reviewers, feel free to point me to parts that you feel are too verbose / superfluous.

File: components/phpunit_bridge.rst [hunk 4]
```diff
@@ -119,7 +119,25 @@ an arbitrary value (ex: ``320``) will make the tests fails only if a higher
...
+When you maintain a library, having the test suite fail as soon as a dependency
+introduces a new deprecation is not desirable, because it shifts the burden of
+fixing that deprecation to any contributor that happens to submit a pull
+request shortly after a new vendor release is made with that deprecation. To
+mitigate this, you can either use tighter requirements, in the hope that
+dependencies will not introduce deprecations in a patch version, or even commit
+the Composer lock file, which would create another class of issues. Libraries
+will often use ``SYMFONY_DEPRECATIONS_HELPER=weak`` because of this. This has
+the drawback of allowing contributions that introduce deprecations but:
+* forget to fix the deprecated calls if there are any;
```
Comment (julienfalque) [inline components/phpunit_bridge.rst]: Does not render as a bullet list because of missing new line.

Comment (greg0ire) [inline components/phpunit_bridge.rst]: Good catch! Fixed!

**Comment by greg0ire** [comment]
Please remove the [WCM] tag and review this

**Review by javiereguiluz — APPROVED** [review]
👍 

Really well explained! Thanks @greg0ire

**Comment by greg0ire** [comment]
Glad you like it @javiereguiluz !

File: components/phpunit_bridge.rst [hunk 5]
```diff
@@ -119,7 +119,27 @@ an arbitrary value (ex: ``320``) will make the tests fails only if a higher
 number of deprecation notices is reached (``0`` is the default value). You can
 also set the value ``"weak"`` which will make the bridge ignore any deprecation
 notices. This is useful to projects that must use deprecated interfaces for
-backward compatibility reasons.
+backward compatibility reasons, or for libraries that do not want deprecations
+triggered by their dependencies to prevent unrelated pull requests from being
+merged, which leads us to the ``"weak_vendors"`` value, described below.
```
Comment (xabbuh) [inline components/phpunit_bridge.rst]: I would revert the changes to this paragraph.

Comment (greg0ire) [inline components/phpunit_bridge.rst]: Why? What's wrong with trying to make a smooth transition like that?

Comment (greg0ire) [inline components/phpunit_bridge.rst]: After more thinking, maybe you're right, I'm kind of explaining the same thing twice here.

File: components/phpunit_bridge.rst (hunk 4 above)
Comment (julienfalque) [inline components/phpunit_bridge.rst]: The list still doesn't render properly, see https://github.com/greg0ire/symfony-docs/blob/weak_vendors/components/phpunit_bridge.rst#making-tests-fail.

Comment (greg0ire) [inline components/phpunit_bridge.rst]: Woops, I think I amended a very old commit and force pushed it.

Comment (greg0ire) [inline components/phpunit_bridge.rst]: Fixed again

**Comment by xabbuh** [comment]
Thank you @greg0ire.
